### DISCUSSION TITLE
[Core Protocol] Retrieving Tokens from L1StandardBridge After Failed Bridging

### DISCUSSION THREAD

**Discussion by ed-iv** [ethereum-optimism/developers#338] [Core Protocol] [answered]
### Did you check the documentation?

- [X] Yes

### Did you read the specs?

- [X] Yes
- [ ] No

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

Suppose an attempt to bridge an ERC20 succeeds on L1 but fails on L2. Funds have been successfully transferred to the `L1StandardBridge` and it is unclear how they can be retrieved from the docs.

The `StandardBridge` contract has some comments saying that in this case funds will be transferred back to the sender, but not clear how that works: https://github.com/ethereum-optimism/optimism/blob/65ec61dde94ffa93342728d324fecf474d228e1f/packages/contracts-bedrock/contracts/universal/StandardBridge.sol#L218-L247

Is it possible to withdraw these stuck funds? Alternatively, if these are transferred automatically back to the sender at some point, what mechanism is responsible for that?

**Comment by sbvegan** [comment]
Hey @ed-iv, sorry for the delayed response. This is a very good question and I'm not sure of the answer. Please let me ask around and see if I can find someone to answer this.

**Comment by sbvegan** [comment, marked as answer]
Hey @ed-iv, I spoke to a smart contract engineer and found out that this is an outdated note. I've opened a [PR](https://github.com/ethereum-optimism/optimism/pull/10073) to remove it from the contract source code. If a token is bridged with no corresponding token contract on the other side, the deposit will get stuck.

**Reply by ed-iv** [reply]
Got it, thanks for the clarification!

**Reply by kayabossq** [reply]
hello @sbvegan im the user of stuck deposit ole , so how to get it back , because i check any address have a finalize withdrawal on this contract : 0x697402166Fbf2F22E970df8a6486Ef171dbfc524

thats a tx have been withdrawal , 
1. https://etherscan.io/tx/0x443a2d50f913d38912559f9314eb5a00e6c6e1a8746379f891d0b6fca53c3c0e
2. https://etherscan.io/tx/0x5561ee3a4c153ec4e9e147cd9de968828c12091f43b1dee8398a7c4a956fa43b

im input on _localToken (address) is smart contract for ETH chain of OLE token 
on _remoteToken (address) , i put BSC smart contract , I should have loaded the blast smart contract, 

the problem is where the token arrived , if i used another bridge they create a new token contracr on L2 , so i can bring back to Layer 1 , 

any suggestions???

**Reply by sbvegan** [reply]
Hey @kayabossq, unfortunately if you've bridged to a non-existent or the wrong contract, the tokens will be stuck. If you're bridging the tokens, its highly recommended to using a bridge UI and to not interact directly with the contracts on a blockchain explorer. I'm sorry for your loss of tokens.

**Reply by Reedlzw** [reply]
I also encountered the same problem.
This is really bad. If a cross-chain transaction fails on the target chain, restrictions should be set on the source chain, or some fallback method should be prepared.
Or the target chain should mint a peged token similar to bridge.dai to avoid faild on L2 
Can we think of a solution together? 
I think it's a bit hard to accept that there is no solution.

**Comment by kayabossq** [comment]
Hello soyboy

I made a question about finalize bridge withdrawal tx.  How they can , ?

Tx on github, the problem if i used another bridge like wormhole they can
create another smartcontract so i can bridge back to L1 , how i can push
this problem , team , blast is ask me to the optimism, bcs they created the
contract ,

Br dika
